File: app/src/main/java/com/junaid/smsapp/utils/MyPreference.java
package com.junaid.smsapp.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class MyPreference {
    private static MyPreference yourPreference;
    private SharedPreferences sharedPreferences;

    public static MyPreference getInstance(Context context) {
        if (yourPreference == null) {
            yourPreference = new MyPreference(context);
        }
        return yourPreference;
    }

    private MyPreference(Context context) {
        sharedPreferences = context.getSharedPreferences("YourCustomNamedPreference",Context.MODE_PRIVATE);
    }

    public void saveData(String key,String value) {
        SharedPreferences.Editor prefsEditor = sharedPreferences.edit();
        prefsEditor .putString(key, value);
        prefsEditor.apply();
    }

    public String getData(String key) {
        if (sharedPreferences!= null) {
            return sharedPreferences.getString(key, "");
        }
        return "";
    }
}

File: app/src/main/java/com/junaid/smsapp/utils/MyPreference.kt
package com.junaid.smsapp.utils

import android.content.Context
import android.content.SharedPreferences

class MyPreference private constructor(context: Context) {
    private val sharedPreferences: SharedPreferences? = context.getSharedPreferences(
        "YourCustomNamedPreference",
        Context.MODE_PRIVATE
    )

    fun saveData(key: String?, value: String?) {
        val prefsEditor = sharedPreferences!!.edit()
        prefsEditor.putString(key, value)
        prefsEditor.apply()
    }

    fun getData(key: String?): String? {
        return if (sharedPreferences != null) {
            sharedPreferences.getString(key, "")
        } else ""
    }

    companion object {
        private var yourPreference: MyPreference? = null
        fun getInstance(context: Context): MyPreference? {
            if (yourPreference == null) {
                yourPreference = MyPreference(context)
            }
            return yourPreference
        }
    }

}